core/base/Dockerfile

```diff
@@ -0,0 +1,57 @@
+# syntax=docker/dockerfile-upstream:1.4.3
+
+# base system image (intermediate)
+ARG DISTRO=alpine:3.14.5
+FROM $DISTRO as system
+
+ENV TZ Etc/UTC
+ENV LANG C.UTF-8
+
+RUN set -euxo pipefail \
+  ; adduser -s /bin/bash -Dh /app -k /var/empty -u 1000 -g mailu app \
+  ; apk add --no-cache bash ca-certificates python3 tzdata
+
+WORKDIR /app
+
+CMD /bin/bash
+
+
+# build virtual env (intermediate)
+FROM system as build
+
+ENV VIRTUAL_ENV=/app/venv
+
+RUN set -euxo pipefail \
+  ; apk add --no-cache py3-pip \
+  ; python3 -m venv ${VIRTUAL_ENV} \
+  ; venv/bin/pip install --no-cache-dir --upgrade --no-warn-script-location pip wheel
```

Comment:
I don't like pip --upgrade here: can't we freeze a version of pip for a specific version of the base image?

Reply:
phrased the other way around: can't we use alpine's version?

Reply:
Alpine's version is also not pinned (apk add).
If you really like to pin the pip version I'll do this via the requirements file.

Reply:
I've added requirements-build.txt to pin this.